### ISSUE TITLE
Enhance a11y for `Calendar`

### ISSUE THREAD

**Issue by hirotomoyamada** [yamada-ui/yamada-ui#5413]
name: Calendar

### Description
4 accessibility violations were found in the `Calendar`.

#### [Components / Calendar / Size](https://yamada-ui.github.io/yamada-ui/?path=/story/components-calendar--size)

Received:
Landmarks should have a unique role or role/label/title (i.e. accessible name) combination (landmark-unique)

Fix any of the following:
  The landmark must have a unique aria-label, aria-labelledby, or title to make landmarks distinguishable

You can find more information on this issue here: 
https://dequeuniversity.com/rules/axe/4.11/landmark-unique?application=axeAPI

#### [Components / Calendar / Shape](https://yamada-ui.github.io/yamada-ui/?path=/story/components-calendar--shape)

Received:
Landmarks should have a unique role or role/label/title (i.e. accessible name) combination (landmark-unique)

Fix any of the following:
  The landmark must have a unique aria-label, aria-labelledby, or title to make landmarks distinguishable

You can find more information on this issue here: 
https://dequeuniversity.com/rules/axe/4.11/landmark-unique?application=axeAPI

#### [Components / Calendar / StartDayOfWeek](https://yamada-ui.github.io/yamada-ui/?path=/story/components-calendar--start-day-of-week)

Received:
Landmarks should have a unique role or role/label/title (i.e. accessible name) combination (landmark-unique)

Fix any of the following:
  The landmark must have a unique aria-label, aria-labelledby, or title to make landmarks distinguishable

You can find more information on this issue here: 
https://dequeuniversity.com/rules/axe/4.11/landmark-unique?application=axeAPI

#### [Components / Calendar / CustomComponent](https://yamada-ui.github.io/yamada-ui/?path=/story/components-calendar--custom-component)

Received:
Landmarks should have a unique role or role/label/title (i.e. accessible name) combination (landmark-unique)

Fix any of the following:
  The landmark must have a unique aria-label, aria-labelledby, or title to make landmarks distinguishable

You can find more information on this issue here: 
https://dequeuniversity.com/rules/axe/4.11/landmark-unique?application=axeAPI

### Additional Information
While it's ideal to cover all tests, if there are too many lines to work on, please create a PR within your capacity.
Any deficiencies not covered by the PR will be automatically created as issues.
